src/Microsoft.AspNet.Identity/RoleManager.cs

```diff
@@ -52,6 +57,7 @@ public class RoleManager<TRole> : IDisposable where TRole : class
             }
 
             Logger = logger?.CreateLogger<RoleManager<TRole>>() ?? new Logger<RoleManager<TRole>>(new LoggerFactory());
+            IdentityLogger = new IdentityLogger() { Logger = Logger };
```

Comment:
Because I need to inject the Logger for the particular class. As you mentioned in the other comment, we should it take it as a mandatory attribute in the constructor